include/zephyr/net/net_ip.h

```diff
@@ -201,23 +201,23 @@ struct sockaddr_in_ptr {
 
 /** Socket address struct for packet socket. */
 struct sockaddr_ll {
-	sa_family_t sll_family;   /* Always AF_PACKET        */
-	uint16_t    sll_protocol; /* Physical-layer protocol */
-	int         sll_ifindex;  /* Interface number        */
-	uint16_t    sll_hatype;   /* ARP hardware type       */
-	uint8_t     sll_pkttype;  /* Packet type             */
-	uint8_t     sll_halen;    /* Length of address       */
-	uint8_t     sll_addr[8];  /* Physical-layer address  */
+	sa_family_t sll_family;   /* Always AF_PACKET                   */
```

Comment:
This is closely related to other stuff in this PR. So while I definitely agree that this should go into a separate commit so that maintainers can pick commits relevant to their code ownership I do not yet understand the point of creating separate PRs (but I'm of course open to change my mind with a good argument). As in the other comment above: If you insist I'll of course follow your advice but would be great if you could give the "why?" together with the "what?".

Reply:
A split commit will do